### ISSUE TITLE
decompiler: support explicitly defining the process a `handle` is related to

### ISSUE THREAD

**Issue by xTVaser** [open-goal/jak-project#1823]
For example
```lisp
(deftype cam-setting-data (structure)
  ((fov degrees)
   (pov-handle handle :offset 16)
   (pov-bone int32)
```
`pov-handle` is always a handle to a `pov-camera` process.  Being able to define it like `(handle pov-camera)` would cut down on a lot of casts.  And hopefully this would be useful in other areas as well.

**Comment by water111** [comment]
agree, we can and probably should do this. It's just annoying to have so many special cases in the decompiler/compiler where dereferencing a field of a certain type just magically gives you a special type.
